### ISSUE TITLE
Does not seem to handle directives with a templateUrl

### ISSUE THREAD

**Issue by qwelyt** [m7r/grunt-ngdocs#187]
I've set up a fresh project using the code provided in this repo. We are trying to set up some 'common components repo' at my company and we thought ngdocs might be a nice piece of software since it allows us to easily see what components actually do before we use them in some new project.

So, my first thought was to add one of our most basic components to the repo and see how it rendered. To my surprise, it didn't really render at all.

I've set up my `app.js` and an `index.html` and on my index page I can see my directive working just as it should ( here is a plunker for it http://plnkr.co/edit/zxrCfABbQqjWcBJKYB78?p=preview )

When I build ngdocs however, the demo part is just plain empty, as can be seen here: http://i.imgur.com/bJmdqE8.png

I also wrote a stackoverflow post about my problem, but so far no answers, here is that link: http://stackoverflow.com/questions/35129405/templateurl-for-directive-when-using-ngdocs

So now I'm wondering; I'm I doing something wrong? Or does ng-docs simply not handle directives with templateUrls?

Also, I might be missing something here, but the only way I can get the demo-part to work is to  write *everything* inside the `<file name="somescript.html">` tag. Which seems rather wrong. The `src` attribute in the `<file>` tag also seems to not work.

This whole ngdocs seems, ironically, very undocumented.

**Comment by blackmjck** [comment]
We've found it necessary in the past to use the `grunt-ngtemplates` tool in combination with ngdocs to wrap the templates up into $templateCache, then include the generated script in the ngdocs array of scripts when setting up the grunt task.

It seems as though anything not explicitly included in the `scripts` array never gets loaded within the documentation for demo purposes, which is probably why you're getting empty spaces.

**Comment by Gargaroz** [comment]
@blackmjck I tried to follow your advice and, since I wasn't abel to find in npm a module called `grunt-ngtemplates`, I thought that must have been a typo and so I installed `grunt-angular-templates`.

The followings are the tasks of `grunt-ngdocs` and `grunt-angular-templates` in the Gruntfile.js:

```javascript
watch: {
      js: {
        files: ['<%= yeoman.app %>/scripts/**/*.js'],
        tasks: ['ngtemplates','ngdocs'],
        options: {
          livereload: '<%= connect.options.livereload %>'
        }
      }
    },
ngdocs: {
       options: {
           dest: 'docs',
           html5Mode: false,
           scripts: [
               'bower_components/angular/angular.js',
               'bower_components/angular-animate/angular-animate.js',
               '<%= yeoman.app %>/scripts/app.js',
               '<%= yeoman.app %>/scripts/commons/main-nav/main-nav.directive.js',
               '<%= yeoman.app %>/scripts/templates.js'
           ]
       },
       api: {
           src: ['app/scripts/**/*.js'],
           title: 'Docs'
       }
    },
ngtemplates: {
      myWebApp: {
        src:      '<%= yeoman.app %>/scripts/**/*.html',
        dest:     '<%= yeoman.app %>/scripts/templates.js'
      }
    }
```

So whenever I save the changes within the ngDocs comments of main-nav.directive.js, both tasks are executed: so far so good, at the end of the ngdocs task I have the desired `templates.js` copied into the directory `docs/grunt-scripts`; **unfortunately the Demo of my directive still doesn't render its template at all**, here's the `@example` of my custom directive:

```javascript
@example
  <doc:example module="mainNavExample">
    <doc:source>
      <div ng-controller="mainNavExampleCtrl as ctrl">
        <main-nav buttons="ctrl.buttonsOpts"></main-nav>
        <p class="h3">Click the buttons to toggle their property</p>
        <div class="btn-group">
          <button type="button" class="btn btn-default" ng-click="ctrl.toggle('share')">Share</button>
          <button type="button" class="btn btn-default" ng-click="ctrl.toggle('clone')">Clone</button>
          <button type="button" class="btn btn-default" ng-click="ctrl.toggle('showAll')">ShowAll</button>
        </div>
        <pre>buttonsOpts: {{ctrl.buttonsOpts | json}}</pre>
      </div>

      <script>
      angular.module('mainNavExample', [])
        .controller('mainNavExampleCtrl', [function() {
         var vm = this;
        
         vm.buttonsOpts = {
           share: true,
           clone: false,
           showAll: false
         };
         vm.toggle = function(item) {
          switch(item){
            case 'share':
              vm.buttonsOpts.share = !vm.buttonsOpts.share;
              break;
            case 'clone':
              vm.buttonsOpts.clone = !vm.buttonsOpts.clone;
              break;
            case 'showAll':
              vm.buttonsOpts.showAll = !vm.buttonsOpts.showAll;
              break;
          };
         };
        }]);
      </script>
    </doc:source>
  </doc:example>
```

Pointing out 2 main facts:
  - the buttons and their bindings are working uber good
  - the `<main-nav>` directive templateUrl is delcared like this:

`templateUrl: 'scripts/commons/main-nav/main-nav.template.html',`

and in the `templates.js` file all the templates are put in the `$templateCache` in this way:

` $templateCache.put('app/scripts/commons/main-nav/main-nav.template.html', " ... actual template ..."`

What am I doing wrong?

@qwelyt Have you eventually found a solution to create the demos for your custom directives?

**Comment by qwelyt** [comment]
@Gargaroz Nope. We dropped ng-docs when we couldn't get the demo part to work. Now we create small demo-pages explicitly for each directive and just use normal comments to document our directives. Since the demo-part never worked for us (we sat for about two and a half week to get it working) it just added workload without giving enough benefits.

**Comment by Gargaroz** [comment]
The struggle was real though.
At that time did you considered the possibility to use tutorial pages together with the API ones, just like the ui-grid guys did to showcase complex directives?

**Comment by qwelyt** [comment]
@Gargaroz We thought about it. But we decided against it. Having both an API documentation as well as tutorial pages is just do much work for our needs. Our directives are small, but combined they are powerful. So a tutorial for how to use it has never really been needed.
What we were looking for was a product-agnostic place to develop our new directives (, services, factories, etc) and showcase them in a simple manner. That was our goal. The API documentation was not top priority. Variable names that explain what they are is documentation enough.
So our goal was to have a unified place to have our stuff showcased. But that never worked for us with ng-docs, so we built our own super-slim demo setup. Only downside is we have to create unique pages for each new thing we do and we don't have auto-generated pages. But it works.
